tom1209-netizen/histolingo-server
# feat: Add all Mongoose models for the project

### Pull Request Title
**feat: Add all Mongoose models for the project**

### Description
This pull request includes the implementation of all the Mongoose models required for the project as per the provided schema diagram. The models created are for the following collections:

1. Players
2. Admins
3. Countries
4. Documentations
5. Roles
6. Topics
7. Tests
8. Questions
9. Feedbacks

### Changes
- Added the `players` model with fields such as `fullname`, `dateOfBirth`, `sex`, `email`, `phoneNumber`, `username`, `password`, `avatar`, `totalScore`, `totalTime`, `rank`, `registrationDate`, `status`, and `locale`.
- Added the `admins` model with fields such as `firstName`, `lastName`, `adminName`, `password`, `roles`, `status`, and `supervisorId`.
- Added the `countries` model with fields such as `name`, `description`, `image`, `status`, `createdAt`, and `updatedAt`.
- Added the `documentations` model with fields such as `source`, `name`, `content`, `status`, `createdAt`, and `updatedAt`.
- Added the `roles` model with fields such as `name`, `status`, `permissions`, `createdAt`, and `updatedAt`.
- Added the `topics` model with fields such as `name`, `description`, `image`, `countryId`, `status`, `createdAt`, and `updatedAt`.
- Added the `tests` model with fields such as `name`, `creatorId`, `documentationIds`, `questionNumber`, `topicId`, `countryId`, `status`, `createdAt`, and `updatedAt`.
- Added the `questions` model with fields such as `topicId`, `dateOfBirth`, `questionType`, `ask`, `correct`, `answers`, `leftColumn`, `rightColumn`, and `corrects`.
- Added the `feedbacks` model with fields such as `createdBy`, `testId`, `content`, `status`, and `createdAt`.

### How to Test
1. Clone the repository and checkout to the branch `feature/admin-model`.
2. Run `npm install` to install all dependencies.
3. Start the MongoDB server.
4. Run the application and ensure all models are correctly defined and connected to the database.

### Checklist
- [x] All Mongoose models created as per the schema diagram.
- [x] Fields and data types are correctly defined.
- [x] Relationships and references between models are properly set up.
- [x] Appropriate validations and constraints are added.

### Related Issues
This pull request is related to the following Jira tickets:
- HA-67: Create model
